scrunch/tests/test_datasets.py

```diff
@@ -439,6 +445,14 @@ def _session_get(*args):
         ds.create_crunchbox()
         ds_mock.boxdata.create.assert_called_with(expected_payload)
 
+    def test_dataset_size(self):
+        """
+        Using `ds.size` actually returns the value of `ds.resource.body.size`
+        """
+        resource = mock.MagicMock()
+        ds = StreamingDataset(resource)
+        assert str(ds.size.rows).startswith("<MagicMock name='mock.body.size.rows' id='")
```

Comment:
Best to load the `resource` mock to have the right attributes (`Mock(size={'rows': 666})`) then you can assert that the size is 666, better than asserting over the Mock's instance `repr`